core/templates/dev/head/pages/library/SearchBarDirective.js

```diff
@@ -43,6 +43,81 @@ oppia.directive('searchBar', [
               }
             )
           );
+          $scope.activeMenuName = '';
+          $scope.ACTION_OPEN = 'open';
+          $scope.ACTION_CLOSE = 'close';
+          $scope.KEYBOARD_EVENT_TO_KEY_CODES = {
+            enter: {
+              shiftKeyIsPressed: false,
+              keyCode: 13
+            },
+            tab: {
+              shiftKeyIsPressed: false,
+              keyCode: 9
+            },
+            shiftTab: {
+              shiftKeyIsPressed: true,
+              keyCode: 9
+            }
+          };
+          /**
+           * Opens the submenu.
+           * @param {object} evt
+           * @param {String} menuName - name of menu, on which
+           * open/close action to be performed (category,language).
+           */
+          $scope.openSubmenu = function(evt, menuName) {
+            // Focus on the current target before opening its submenu.
+            angular.element(evt.currentTarget).focus();
+            $scope.activeMenuName = menuName;
+          };
+          $scope.blurNavigationLinks = function(evt) {
```

Comment:
Where is this function being used ? I could not see this function being referred to anywhere in the html file.